keras_nlp/utils/text_generation_test.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for Text Generation Utils."""
+
+
+import tensorflow as tf
+
+from keras_nlp.utils.text_generation import greedy_search
+
+
+class TextGenerationTest(tf.test.TestCase):
+    def setUp(self):
+        super().setUp()
+        vocab_size = 10
+        feature_size = 16
+
+        # Create a dummy model to predict the next token.
+        model = tf.keras.Sequential(
+            [
+                tf.keras.Input(shape=[None]),
+                tf.keras.layers.Embedding(
+                    input_dim=vocab_size,
+                    output_dim=feature_size,
+                ),
+                tf.keras.layers.Dense(vocab_size),
+                tf.keras.layers.Softmax(),
+            ]
+        )
+
+        def token_probability_fn(inputs):
+            return model(inputs)[:, -1, :]
+
+        self.token_probability_fn = token_probability_fn
+
+    def test_generate_with_1d_prompt(self):
+        inputs = tf.constant([1])
+        outputs = greedy_search(self.token_probability_fn, inputs, max_length=5)
+        self.assertEquals(outputs.shape, [5])
+
+    def test_generate_with_2d_prompt(self):
+        inputs = tf.constant([[1], [1]])
+        outputs = greedy_search(self.token_probability_fn, inputs, max_length=5)
+        self.assertEquals(outputs.shape, [2, 5])
+
+    def test_generate_with_list_prompt(self):
+        inputs = [[1], [1]]
+        outputs = greedy_search(self.token_probability_fn, inputs, max_length=5)
+        self.assertEquals(outputs.shape, [2, 5])
+
+    def test_generate_with_ragged_prompt(self):
+        inputs = tf.ragged.constant([[1], [2, 3]])
+        with self.assertRaises(ValueError):
+            greedy_search(self.token_probability_fn, inputs, max_length=5)
+
+    def test_assert_generation_is_correct(self):
+        def token_probability_fn(inputs):
+            batch_size = inputs.shape[0]
+            prob = tf.constant([[0.1, 0.2, 0.3, 0.4]])
+            return tf.repeat(prob, batch_size, axis=0)
+
+        batch_size = 10
+        inputs = tf.ones([batch_size, 1], dtype=tf.int32)
+        outputs = greedy_search(token_probability_fn, inputs, max_length=3)
+        self.assertAllEqual(outputs[:, -1], tf.repeat(3, batch_size))
+
+    def test_end_token_id(self):
+        def token_probability_fn(inputs):
```

Comment:
shouldn't we test a token_probability_fn that actually returns end_token_id here? rather than a prompt that contains end token id?

and don't we want to test end_token_id != pad_token_id to fully test this logic?

Reply:
1. It's the same? Here we want to test that when end_token_id is encountered, all subsequent tokens should be masked out. The logic that "next token is properly appended to prompt" is covered in other cases.
2. SG!

Reply:
> It's the same?

A utility that only checked the prompt for eos would pass this (and all) unit tests, while being obviously incorrect. But not a big deal!